### ISSUE TITLE
Coordinating the next release

### ISSUE THREAD

**Issue by pjbroad** [raduprv/Eternal-Lands#137]
We have a full client update coming that includes maps updates (from @feeltheburn) and a server update (from @raduprv).  This will be the next major version bump.  We've been holding off major changes to the client code so but have the exciting prospect of #132 (from @gvissers) so we need to decide whether to merge that so we can test fully before the release.  I'm in favour of merging #132 but what do others think?  Are there other changes should we consider?

**Comment by raduprv** [comment]
Before we do this, we must compile the Android version for 64b, or else Google won't allow it on google play.
And the maps were not tested at all on the test server.

**Comment by gvissers** [comment]
As I said elsewhere, whether we merge the encryption code should IMO depend on how much time we have for the next release. If there's still a month or two left, merging now will give us enough time to get this tested properly and work out the remaining kinks, I think. There's probably still some work to be done, as I have tested this only lightly on Windows, and not at all on MacOS or Android (and FreeBSD? does anyone use that? Maybe I should set up a VM). Worst case, compiling without USE_SSL should give you the original network code, and the other new code from #132 will then never be used.

**Comment by pjbroad** [comment]
> Before we do this, we must compile the Android version for 64b, or else Google won't allow it on google play.
> And the maps were not tested at all on the test server.

Would this be a version compiled from the code base I've been working on (the android branch)?  If so, a 64b version is already included and can be made the only version included if that is what's required.

> As I said elsewhere, whether we merge the encryption code should IMO depend on how much time we have for the next release. If there's still a month or two left, merging now will give us enough time to get this tested properly and work out the remaining kinks, I think. There's probably still some work to be done, as I have tested this only lightly on Windows, and not at all on MacOS or Android (and FreeBSD? does anyone use that? Maybe I should set up a VM). Worst case, compiling without USE_SSL should give you the original network code, and the other new code from #132 will then never be used.

Are you building the windows version using the msys environment setup I also use?  I recently installed FreeBSD in a VM so that I could get the cmake file work for BSD; there's a example cmake line in the CMakeList.txt comments. The latest main branch works fine on FreeBSD and I could easily try the encryption version on the same VM if you like.

**Comment by raduprv** [comment]
>Would this be a version compiled from the code base I've been working on (the android branch)? If so, a 64b version is already included and can be made the only version included if that is what's required.

I am not sure exactly, I have to check with google. Is there a way to pack both 32b and 64b in the same apk? Because a lot of older phones are 32b only.

>On the maps, didn't mean to imply they had been tested and understand form Burn they will be available to test soon.

Yes, I know, I just stated that before we can think of an update we need to test the maps properly. They are on the test server now.

**Comment by feeltheburn** [comment]
On the map end, I've been doing some thorough mapwalk checking of every change, still working on that. Also putting together a bundle of new elms, updated tab maps, and a list of changes for people with a separate client set up for Test server so they can check them out as well. 
(Dropping that on the general population for testing in the past has only led to people screwing up their main client...)

Thanks to testing scripts I created (Learner uses them on OL maps and helped with a few tweaks of them), I was able to test and fix a lot of stuff that map makers in the past would not have caught. 
You can see a compiled result of tests on one map here for example, it also explains what the marking is: https://el-db.com/tempstuff/map6nf_cave-test-results.png 
That's actually a combined image. My script creates a PSD file and each of those numbered tests are on separate layers so they can be looked at individually. The scripts actually test a lot of stuff not seen there as well.

So it's mostly just a visual thing now, and human testing of all use_areas and teleports to ensure they go to the right place, and if new/changed places are too cluttered for play, or any timed/walk damage is too much/little.  Things that can't always be determined. 



Thus far the fixes I've found that need to be done are minimal, nothing game-breaking.  Should give testing a week or two at least, with at least one update with fixes of issues I've found coming as soon as I feel the changes are fully tested and I've caught everything I can.

My opinion is we could aim for an August release, but if we aim for September we could give more time for the above-mentioned issues including the possibility of use_ssl to be implemented, and I could get some additional graphics work done (making more unique inventory item images for example for stuff sharing the same image like I did last time for example). Given how long it's been, I don't think September would be unreasonable.

I'm done with the maps though, other than fixing stuff found in test of course. So my part including testing could be done by August if that's preferred.

**Comment by gvissers** [comment]
> Are you building the windows version using the msys environment setup I also use? 

Yes. Thanks for setting up detailed instructions, BTW, they were really helpful.

> I recently installed FreeBSD in a VM so that I could get the cmake file work for BSD; there's a example cmake line in the CMakeList.txt comments. 

I did the same here. I don't know what I did wrong the first time around, ended up struggling with finding Cal3D and building and patching my own from source. But it works fine (after a small #include fix) with system Cal3D and the line suggested in CMakeLists.txt, so I must have messed something up the first time. Could be that I was working from an old branch. The SSL code also works without further adaptations, so that all seems in order. (it's slow though in the VM, I get max 8FPS inside a building, 2 FPS outside, I'll have to see if I can tweak the VM settings to help that. Even windows is better). 

I'll be on vacation in the first two weeks of August, so if we include the encryption I'd prefer to release after that, i.e. second half of August/September. I'm in favor of merging it, though (totally unbiased of course). Perhaps after we check if it works on Android, but I don't know how to build for that.

**Comment by pjbroad** [comment]
> Perhaps after we check if it works on Android, but I don't know how to build for that.

There's an Android branch which I just updated.  You could try using [these build instructions](https://github.com/pjbroad/el-build-methods/tree/master/android).  I've got to update the assets download for 1.9.5p9 though the previous version should work OK.

**Comment by gvissers** [comment]
> There's an Android branch which I just updated. You could try using [these build instructions](https://github.com/pjbroad/el-build-methods/tree/master/android).

Gaahh... how many goats did you have to sacrifice before you had all that working!? Thank goodness you already worked this out, because I never would have.

I set up a clean ubuntu install in a virtualbox VM, and followed the rest of your instructions. The ssl branch merges cleanly, and builds with a small patch **without** USE_SSL (and as I'm typing this I see there's two typos in the error message we might want to correct as well):
```
diff --git a/events.c b/events.c
index 655870f2..43fbeaad 100644
--- a/events.c
+++ b/events.c
@@ -257,7 +257,7 @@ int HandleEvent (SDL_Event *event)
 #ifdef ANDROID
                case SDL_APP_DIDENTERFOREGROUND:
                        SDL_Log("App returned to forground");
-                       if (disconnected && !locked_to_console)
+                       if (is_disconnected() && !locked_to_console)
                        {
                                SDL_Log("Reconnectecing after return to forground");
                                connect_to_server();
```

Unfortunately I have as of yet not been able to get OpenSSL to build in your build environment. I'm sure it can be done, but I can't seem to get the paths right. So as of yet, I haven't tested the new network code on android yet.

**Comment by feeltheburn** [comment]
So would an aim for a "first half of September" for a release be reasonable? Latest end of September if unforeseen issues arise or more time is needed for the SSL thing?



**Update on my side:**

**- Maps are looking good.** Several have been on and tested the new map areas. Only a handful of minor issues found, all text (like text that gets shown when you use a door or such), so pretty good to go. I'll send a minor update pack to radu in a week or so once others have had more time to explore the new areas and ensure no further issues.

**- New Images for Inventory.** With thoughts of a late august release earliest (since gvissers will be on vacation the first part of august), I went ahead and started a load of work on inventory images. Specifically, doing something that's long been requested, the ability to differentiate graphically between a weapon/armor and its second-hand/used/degraded/pre-owned counterpart, as they currently look the same in inventory. Working on both those and the updates needed on the server's items.def file to change the graphics for the degraded items.
(I showed a few samples of this in @@6, and got cheers. It's been wanted for as long as I've been playing.)

**- Possible to do with a September release:** Also looking at unique images for the "special" forms of items (ex: Tit cuisses of cooldown removal vs. regular Tit cuisses) as they also share the same images.
Basically if it's not a book (too many of those), if I can make a unique image for it, I'm going to try to make one.

**Comment by gvissers** [comment]
I finally managed to build a client with SSL support on Android, and what a _fun_ process that was :grimacing: :sweat: . Anyway, I have sent out pull requests for the affected repositories (Eternal-Lands/android branch, and el-build-methods). I have verified that it works on my phone, both encrypted and unencrypted, but it has only been tested on the one device (I believe 64bit, but I'd have to check to be sure).

Since it seems to work on all platforms I've tried (Linux, Windows, FreeBSD, Android), I will merge the SSL branch soonish (tomorrow, maybe wednesday) if there are no objections.

@feeltheburn , I've been meaning to test the maps, but haven't found time yet.

**Comment by feeltheburn** [comment]
That's okay grum, I think we've got enough testers, including explorers that know the maps well like angler. But there's still time so send input if you have time later and see any issues.

**Comment by gvissers** [comment]
I have merged #132 and enabled USE_SSL by default so that the code can get wider testing. Things should work as they were on unencrypted connections. Encrypted connections can be tested on Learner's proxies, just add the following lines to servers.lst:
```
lrnr-ssl-main   main            proxy1.other-life.com   3000    encrypted   Encrypted proxy to the main game server
lrnr-ssl-test   test            proxy1.other-life.com   3001    encrypted   Encrypted proxy to the test server
```
Using one of these will pop up a warning that the certificate could not be verified because these aren't distributed with the client yet. For now, just ignore the warning and click "Continue".

**Comment by feeltheburn** [comment]
Update on my end:

- Maps - seem to have gotten some decent testing and feedback. No game breakers, just some minor tweakage. I'll do that over the weekend and send radu the "beta" map fixes for Test around Monday (give or take a couple days).

- Inventory Images - I've made images to handle all "broken/second-hand/damaged/used" weapons/armors, and rare versions of weapons/armor, as well as other stuff that was sharing the same images, fixing a long-time-coming complaint of objects sharing the same image. (Some things like Books, Removal stones, and such will continue to share images as there's too many and they're too limited to bother with.)

- Creatures - I have made a few new looks to differentiate some of the creatures sharing the same look. Assisted by grum in getting them set up properly in the client, Wizzy doing some hidden invading of some to test that don't exist on Test server, and people who have seen screenies of them have so far been positively responding. They'll of course need approval, but the players seem to like the test images so far.

**Comment by gvissers** [comment]
Learner has indicated that he prefers to support both encrypted an plain connections on the same port. So I have had to adjust both the proxy code and the client code to enable this. Somewhat ironically, I have reverted back to the original design, where the client sends a LETS_ENCRYPT packet to the server on startup, and the server (or proxy, rather), responds with a LETS_ENCRYPT packet of its own. If the server says it does not support encryption, or when no response comes within 5 seconds, the connection is closed. 

This is somewhat of a big change somewhat late in the game, so I would really like it if people could test this. As indicated previously, I am currently on vacation, so I may be somewhat slow to respond, but I will follow this issue, and can be reached through the usual means (forum PM ought to work, probably won't be in game though).

**Comment by feeltheburn** [comment]
I think there's a few issues with black screens and such on the forum that are holding the release for now, but just for my part I'm gonna put a stop to anything I'm doing by mid-September, so if something's not done it just won't go in. I'll pack all the maps and other stuff not in this git up and have it ready for release at that point.

**Comment by bendoughty** [comment]
Everything seems to be working great. I have created a pull request (#152) with an updated Xcode project file.

**Comment by feeltheburn** [comment]
Running behind on my end, was sick for almost a week among other things.

Sent some final testing maps to radu just now, should be good at this point. I got some tab map work to finish up but it's mostly done, then I'll ball up the pack of data files to use for final 1.9.6 builds and set them up to be downloadable for client builders.

**Comment by feeltheburn** [comment]
Actually just tested a final map issue that I sent to Radu yesterday. On my end, I'm final-touching the tab maps (updating the full c1/c2 continent maps that you see in the Tab map view when you click on the upper left corner map image is a bit of a task I forgot I needed to do), then make sure all the updated data files (everything not in this git) are gathered and zipped up for those doing the compiling to download and use.

I shouldn't need more than a week at most, but likely by this weekend.


I'd imagine the eye candy problems ( http://www.eternal-lands.com/forum/index.php?/topic/61631-issues-with-eye-candy-for-example-overly-intense-or-window-darkness-after-cast/ ) are more of a release slowdown issue? Happening in the middle of an invasion, that's the worst time to be getting a bug like that.

**Comment by pjbroad** [comment]
Looks like we found and fixed the eye candy issue so we're ready for the final stages.

The Android client source is up to date with the main client.  @raduprv if you're not doing to build it yourself, could you resign or unpack and resign a build I do for android?  I also have have build instructions and scripts in [this repo](https://github.com/pjbroad/el-build-methods) that you might want to try.  The Android client now supports CUSTOM_UPDATE and CUSTOM_LOOK which you may want to turn off for the play store.

**Comment by raduprv** [comment]
I guess I can give you the key thing, it's mainly used for the EL binary anyway (and some other app I have on the Playstore that I don't care very much about).

**Comment by feeltheburn** [comment]
I'm almost done. Bluap, check forum PM.

I'll have the server change list sent to radu later today (not much to do), and updated data file pack available for download as well.

Er... that android build needs these new data files, too. ;) 

It's a combo of the private el data git which is getting a last minute update, the private map file git, and Odie's encyclopedia git. Git 'er done...

**Comment by pjbroad** [comment]
> I guess I can give you the key thing, it's mainly used for the EL binary anyway (and some other app I have on the Playstore that I don't care very much about).

That's fine. Not sure of the best way to send it to me but you have my email.

On the SSL stuff.  It's been 100% stable for me and I know Aislinn has been using it too.  Can we at least included the server certificates in the data pack?  My preference would be to also add additional lines to the server.lst file too, but not necessarily enable it as the default.

**Comment by raduprv** [comment]
Yes, we can introduce the certificates, but not the server list for now. Those who wish to change it can do it by hand, but I don't want it to be available for everyone just yet.
Btw, the Android APK needs to be under 100mb in size, or else we can't host it on Google Play. So some non essential tab maps will have to be removed or downsized.

**Comment by feeltheburn** [comment]
Right. Based on what bluap's current build size is using standard tab maps, the special reduced-size tab map pack I have put together just for Android will put it well below that without removing any. So that's covered.

**Comment by feeltheburn** [comment]
bluap - As soon as I get the certificates in the data git (or otherwise if they shouldn't be there for some reason) I'll post the data pack and the android map pack.

**Comment by pjbroad** [comment]
> bluap - As soon as I get the certificates in the data git (or otherwise if they shouldn't be there for some reason) I'll post the data pack and the android map pack.

Thanks, merge request done.

**Comment by feeltheburn** [comment]
**DATA PACK**

https://el-db.com/tempstuff/196release/el_data-196.zip


**ANDROID MAP PACK**

https://el-db.com/tempstuff/196release/maps-196-FOR-ANDROID-ONLY.zip

IMPORTANT: For Android, it includes everything needed so completely delete the /maps/ folder in the data pack and replace it with this, don't just overwrite. Some unusable-by-player maps don't appear in the android pack that do in the data pack, overwriting would keep them and their larger tab maps.

**Comment by gamil-zirak** [comment]
> **DATA PACK**
> 
> https://el-db.com/tempstuff/196release/el_data-196.zip
> 

The cyclops_2.xml file is missing causing an error preventing the game from loading.  The other new actor definitions are present though.

**Comment by feeltheburn** [comment]
FIXED. There is no cyclops_2 files, that was something I was testing but didn't finish.

You can redownload, or open el_data/actor_defs/actor_defs.xml, find the *2* lines that reference cyclops_2, and delete them.

**Comment by gamil-zirak** [comment]
Thanks, that fixed it.

**Comment by pjbroad** [comment]
It's minor I know but I was just comparing the data zip with the versions in the client dev directory and noticed in item_info.txt Turquoise is missing the second "u".

It's probably time to bump the version numbers for the client and build some test packages.  @raduprv do you want to up the server version numbers? Currently version_first_digit=10 and version_second_digit=29 (or 28 if compiled without SSL).

**Comment by raduprv** [comment]
version_second_digit=29 sounds good.

**Comment by pjbroad** [comment]
> version_second_digit=29 sounds good.

Its already 29 if compiled with USE_SSL (the default), but 28 if compiled without USE_SSL.  So leave it at that or bump both to 29?

**Comment by raduprv** [comment]
28 is the current one, so they both must be upgraded to 29

**Comment by feeltheburn** [comment]
The item_info is script-built directly from the items.def file that's on the server. Which has always been without the second U according to my harv stats. I didn't wanna touch that because it'll mess with counters.

(Spelling-wise in the server items list, I only fixed 2: "raccoon hat" got its missing C because it's not something that'll be high on anyone's counters, if it's on them at all, and "Wilhelm Hood Removal" stone because it was impossible to search for with an L missing in the name, and there's no counter issues.)

![turq](https://user-images.githubusercontent.com/2794779/140009519-0bd1620e-4ebe-4c2f-b1d1-86007f549f04.jpg)

**Comment by pjbroad** [comment]
> The item_info is script-built directly from the items.def file that's on the server. Which has always been without the second U according to my harv stats. I didn't wanna touch that because it'll mess with counters.

I only mention it as the previous item_info.txt file shipped with the client spelt it correctly.  The description in that file is not used by the counters as they scrape the name from the server messages "you made a <thing>".  It not a big deal.

**Comment by raduprv** [comment]
Bluap, which e-mail address shall I use to send you the key file? I have 3 addresses for you, not sure which one is your primary one.

**Comment by pjbroad** [comment]
> Bluap, which e-mail address shall I use to send you the key file? I have 3 addresses for you, not sure which one is your primary one.

I've sent you an email.

**Comment by feeltheburn** [comment]
@pjbroad bluap, have you tested the Android build size with the special tab maps for it?

**Comment by pjbroad** [comment]
> @pjbroad bluap, have you tested the Android build size with the special tab maps for it?

Just tried it, apk size is 89M so its fine.

**Comment by feeltheburn** [comment]
Okay, coolness. They should be okay but holler if you test it and the tab maps aren't showing well. We can try some larger (pixelwise) jpg versions that might be better. 

...


I've posted randomly multiple times in @@6 to try and get people to test the eye candy issue over several days, don't think we're gonna get any more to check it. Those of us that have tested it are all inclined to think it's fixed.

The client otherwise is very well tested with people actually using bluap's builds or like me self-compiling the git and using it in game play daily.

The updates I sent radu for server was a quite short list that shouldn't take long to do at the last minute.


Are we waiting on anything else at this point? Can we finally do this? ;)

**Comment by pjbroad** [comment]
On Android the tab maps are find, not as shape but still readable.


> Are we waiting on anything else at this point? Can we finally do this? ;)

Looks like it, I'm doing RC build and will upload them shortly.

**Comment by feeltheburn** [comment]
Okay, if they aren't good enough we can try again later with an updated build for it with some better quality images that'll keep the build below 100M.

It's the pixel-size limitations of dds files that make me wanna switch to jpg. We have 1024x1024 in the main client, the android ones are 512x512. There's no in-between for dds. With jpg we could do 800x800 or so and likely stay well under the client size limit.

But if they work as-is, we can discuss that later. I'd rather get the client out.

**Comment by pjbroad** [comment]
@feeltheburn Just realised the data pack does not contain the standard TTF fonts.  I'll do a pull request containing them now.

**Comment by feeltheburn** [comment]
DOH! Okay, I'll get a new pack out shortly.


(For anyone running an rc build: Keep in mind that they won't show the updated inventory images on any server, as they haven't been updated server-side.)

**Comment by pjbroad** [comment]
Merge request done.

**Comment by feeltheburn** [comment]
https://el-db.com/tempstuff/196release/el_data-196.zip  

Data pack is updated.


@gamil-zirak - note the data update. The change is the added /fonts/ directory in the data pack, so you only need add that to any build you're doing.

**Comment by raduprv** [comment]
Before the update, I'd like for many people to test the binaries for at least a week. This also gives me some time to settle some personal issues (my dad has covid and he is not very well).
So once we have the binaries for all the platforms, we should post the links on the forum (after I upload them).

**Comment by feeltheburn** [comment]
These binary tests would have to still be done with 1.9.5's data files, of course, the map differences would be very problematic otherwise (unless used on Test server only). So just a binary file that can be plopped in to replace the user's current one, or a full build using the old 1.9.5 data.

Many have already been using bluap and Ben's updated builds for quite a while so I doubt we'll see any problems.


But of course we can hold off and plan for next weekend (tentatively) for the personal issues regardless.

**Comment by raduprv** [comment]
Yes, we can ask people to test the new binaries with the old data. Every small change can cause some issues. The new Android client should also be tested, not a lot of people tested it before.

**Comment by pjbroad** [comment]
@raduprv sorry to hear about your father, I hope he gets better soon.

I'm currently building and uploading packages to the [1.9.6.RC release](https://github.com/raduprv/Eternal-Lands/releases/tag/1.9.6.0-rc1) on github.  I've included full packages with the new data pack and just executables for Linux and executable+dll packages for Windows.  I'll do the Android build next then add some explanatory text to the release information.

**Comment by brunoramoslu** [comment]
I have tested the .sh installer on a clean installation of Debian and there is a bug when /home/user/.local/share/applications folder does not exist the script is installing the desktop file directly in /home/user/.local/share/applications.
This means that location is not a folder but instead the .desktop file itself.
I'm writing this on my mobile so cannot attach a log.

![20211107_161817.jpg](https://user-images.githubusercontent.com/3391955/140651040-8713305f-7399-421a-b850-6c0102b0f6ab.jpg)

**Comment by brunoramoslu** [comment]
Not sure this is related to the release, but in the test server #ach is returning an error.

Unkown: #ach

**Comment by feeltheburn** [comment]
> Not sure this is related to the release, but in the test server #ach is returning an error.
> 
> Unkown: #ach

The Test server is VERY far behind the main server, I think it's from the days of 1.9.3. 

#ach came after that, so that's an expected result. As long as it works on Main, it's okay.

**Comment by gamil-zirak** [comment]
One heads-up for anyone that's running a 1.9.6 binary with the 1.9.5 data: any custom or unofficial files in ~/.elc/updates/1_9_5 will no longer be loaded. If you're inviting other's to test, it may be a good idea to give them a warning that this is expected and not a bug.

**Comment by pjbroad** [comment]
> I have tested the .sh installer on a clean installation of Debian and there is a bug when /home/user/.local/share/applications folder does not exist the script is installing the desktop file directly in /home/user/.local/share/applications.

Thanks for finding that, I've fixed the script and uploaded a new version of the installer package.

**Comment by feeltheburn** [comment]
> One heads-up for anyone that's running a 1.9.6 binary with the 1.9.5 data: any custom or unofficial files in ~/.elc/updates/1_9_5 will no longer be loaded. If you're inviting other's to test, it may be a good idea to give them a warning that this is expected and not a bug.

We may wanna tell them to rename 1_9_5 to 1_9_6.

That would be good for testing as well, making sure custom files still work.

**Comment by gamil-zirak** [comment]
> We may wanna tell them to rename 1_9_5 to 1_9_6.
> 
> That would be good for testing as well, making sure custom files still work.

I'd done that so that I wouldn't lose my nice Burned maps. I can confirm that custom files still work.

**Comment by pjbroad** [comment]
All the release candidate builds for Linux, Windows and Android are done.
See the [1.9.6.0-rc1 release](https://github.com/raduprv/Eternal-Lands/releases/tag/1.9.6.0-rc1).

**Comment by feeltheburn** [comment]
@bendoughty Can we get a mac rc build (with 1.9.5 data, just update the binary)?

We need to emphasize only binary updates for using on Main, and any with 1.9.6 data should be marked for use on Test server only. (Cleanup work of height tiling is done on pretty much every non-instance map and **will** result in mapwalking hell, on almost every main map, not just the stuff listed in the change log post.)

**Comment by bendoughty** [comment]
@feeltheburn Certainly, though unfortunately I am currently away and the earliest I will be able to get this done is Friday morning.

So I don’t leave you entirely empty handed until then: Feedback on the interim builds I share with my inner circle has been good, and I haven’t had any reports of problems.

**Comment by feeltheburn** [comment]
With that then, perhaps go ahead and announce the test binaries using 1.9.5 data for Win and Linux and give it a week or so, announce Mac ones on Friday, which'll still give them a few days?

**Comment by raduprv** [comment]
I posted on the forums: http://www.eternal-lands.com/forum/index.php?/topic/61661-eternal-lands-196-release-candidate/

**Comment by gvissers** [comment]
I have added a last minute change at the request of Learner: certificates can now also be loaded from the updates directory, allowing updates of the server certs without troubling the user (if they enable auto-updates). It's not worth rebuilding the test binaries over, though, as it only adds a single directory to the list. This should be merged into the OL client branch at some point, though.

**Comment by pjbroad** [comment]
You may have seen a few commits mostly fixing scaling issues for the language, rules and new character windows.  I've also hopefully improved how the on-screen keyboard is opened on Android; fixing a chat window issue in passing.  I really hope that's it and we can release soon.  Do we want another RC?

**Comment by raduprv** [comment]
At least for Android, yes, because it's harder to do updates for it.

**Comment by feeltheburn** [comment]
'Tis been quiet of late... Will we be able to make 1.9.6 and early Xmas bonus? ;)

**Comment by raduprv** [comment]
The Android client was what set us back a little. Now that it seems to be solved, we can resume the process.
Bluap, can you please give me a final Android version for release? I will upload it on Playstore, and see if there are any issues.

**Comment by feeltheburn** [comment]
Yeah, I just wasn't sure if the issues were resolved. Been watching the forum threads.

Yay, hopefully time to rock this, then. :)

**Comment by pjbroad** [comment]
> The Android client was what set us back a little. Now that it seems to be solved, we can resume the process. Bluap, can you please give me a final Android version for release? I will upload it on Playstore, and see if there are any issues.

I was just squeezing in some improvements to the window positioning and scaling, mostly for Android.  That's done now.  There does appear to be an issue that backup is not working on at least newer Android versions (I've not changed the manifest file settings).  I'd like to see if I can fix that, probably this weekend, but I'll aim to do builds anyway if I can't.  Hopefully we can wait another few days?

**Comment by raduprv** [comment]
Sure, no problem, take your time!

**Comment by pjbroad** [comment]
> Sure, no problem, take your time!

I've uploaded the Android RC 5 build to github.

**Comment by pjbroad** [comment]
Shall we do the release? If agreed, I put the release label in place and do a final set of builds.

**Comment by raduprv** [comment]
Yes, I think we are ok for a release! Please let me know after you finished all the builds.

**Comment by pjbroad** [comment]
All the Windows and Linux build are available on the [release page](https://github.com/raduprv/Eternal-Lands/releases/tag/1.9.6.0).

@bendoughty let me know when have build against the 1.9.6.0 tag then I'll add it it the page.

**Comment by pjbroad** [comment]
The API change is fine.  I set the manifest "versionName" to "1.9.6.0 release" to match the desktop client.  AFAIK its is just a string used in the application information. If we change the manifest, the client version will still be 1.9.6.  Is 1.1.3 your preference or a play store requirement?

**Comment by raduprv** [comment]
It's a requirement, you can't have two apks with the same version.

**Comment by raduprv** [comment]
Oh, I mean it is better to increment it to the next number I used for the Android version, because we might have an update to it if there are issues and the Android updates are separate from main client updates.

**Comment by pjbroad** [comment]
I think the actual version number in the manifest is is the "versionCode" and integer.  I didn't know what you last set set to so I've updated it to 10 which allowed me to manually install my APK over the playstore version.  I think that is the value that needs updating and so should be fine as is.  The "versionName" is just a string so we can add what ever additional information to it we wish should a separate update be needed.  If at all possible, I'd rather not have to modify the client source then maintain use a special version number for the android client. I'd really like to merge the android branch into master after this release.

**Comment by raduprv** [comment]
Oh, yes. I had version code 6 in the last release.
However, in the google play console I see the versionName.

There is no need to modify the source to show the android specific version in the client. It is more for us. But I guess we can try the version you put (1.9.6) and see if there are any issues.

**Comment by pjbroad** [comment]
OK, thanks.  I've updated the [APK file](https://github.com/raduprv/Eternal-Lands/releases/download/1.9.6.0/EternalLands-Android_1.9.6.0.apk).  The only change is "targetSdkVersion=30" in the manifest.

**Comment by raduprv** [comment]
Thanks!
New problem now :/
You uploaded an APK with an invalid signature (learn more about signing). Error from apksigner: ERROR: MIN_SIG_SCHEME_FOR_TARGET_SDK_NOT_MET: Target SDK version 30 requires a minimum of signature scheme v2; the APK is not signed with this or a later signature scheme

**Comment by pjbroad** [comment]
OK.  I can see the same error if I try to verify the signature.
```
DOES NOT VERIFY
ERROR: Target SDK version 30 requires a minimum of signature scheme v2; the APK is not signed with this or a later signature scheme
```
So I should know if I have fixed it.  I'm trying to find out more.

**Comment by pjbroad** [comment]
I've updated the APK file. The only change is I've resigned with apksigner and it now verifies OK:
```
Verified using v1 scheme (JAR signing): true
Verified using v2 scheme (APK Signature Scheme v2): true
Verified using v3 scheme (APK Signature Scheme v3): true
Verified using v4 scheme (APK Signature Scheme v4): false
Verified for SourceStamp: false
Number of signers: 1
```
Hopefully that fixes the error.....

**Comment by raduprv** [comment]
Great, I released the new version on Android, waiting for Google's review.

**Comment by pjbroad** [comment]
Fantastic!

**Comment by bendoughty** [comment]
@pjbroad I'll get to work on the macOS build now, shouldn't be too long. I'll send you a link to the app bundle and updated framework pack as soon as I am done. 😊

**Comment by feeltheburn** [comment]
> Great, I released the new version on Android, waiting for Google's review.

Is this complete with 1.9.6 data/maps?  Which means all the other clients get updated with the server and...?

Zomg a release? :-D 


I'll get the moderator invasion map info page updated for them for when the update happens. (Nothing to wait on here.)

**Comment by pjbroad** [comment]
Just a quick sanity check.  On the test server, it looks like the item images ID have not yet been update.  I presume that will be done at some stage and for the main sever too when that gets its update for the release.

**Comment by bendoughty** [comment]
@pjbroad The macOS build is done and the links are on their way!

**Comment by pjbroad** [comment]
> @pjbroad The macOS build is done and the links are on their way!

Great, all uploaded thanks.  I think we have all the builds ready. :)

**Comment by raduprv** [comment]
Once Google approves the Android update, I will upload the new clients on the website and link them on the download page.
Then I will wait a few days for people to download the new clients, and have a chat with Burn to make sure I properly update the maps and other stuff on the main server.

**Comment by feeltheburn** [comment]
Everything you need is in the list I sent in e-mail: all map elm/def files in the zip pack, update the items.def file, update the pear map list, and the ice/blue spawns for the new locations made for them.

But except when I'm asleep you can reach me on FB any time if there's any questions or issues with those.

**Comment by raduprv** [comment]
I had to resubmit it for review, because google wanted a test character (they couldn't bother to create a new one).

**Comment by pjbroad** [comment]
I don't think its a critical bug but the the [spells window bug](http://www.eternal-lands.com/forum/index.php?/topic/61508-new-development-version-of-the-android-client/&do=findComment&comment=597869) for android is now fixed.  I've published a 1.9.5 dev package update but not the release version.  @raduprv if you still need to submit the package again, I can build you a new version.  I should probably build the new version and replace the one on the releases page anyway.

There is also a minor bug with the default window position for the spells window.  That's fixed too but its probably not worth a full rebuild or all the packages.  Perhaps we could wait a week or so after release then publish an update if there are a few more bugs that emerge once more people are using it.

**Comment by raduprv** [comment]
If they approve it, I won't resubmit it for now, but we can always do an update later on to the Android version.

**Comment by pjbroad** [comment]
I see the android version is approved, that too longer than I would expect.  It's good though.
I've added a new [android build](https://github.com/raduprv/Eternal-Lands/releases/download/1.9.6.0/EternalLands-Android_1.9.6.0-1.apk) to the release assets with a "1.9.6.0-1" tag.  I've not deleted the "1.9.6.0" version you uploaded to the play store.  If you add the apk to the downloads page, you could use the "1.9.6.0-1" version.

**Comment by raduprv** [comment]
Ok, great, thank you all! I am uploading the binaries on the web server now, will modify the download page after they are done.
And we can have a server update in a few days.

**Comment by pjbroad** [comment]
Are we OK to close this issue?

**Comment by gvissers** [comment]
Ok with me, unless there is still a server update in the works that needs coordinating?

**Comment by feeltheburn** [comment]
Nah, this can be closed. Next server update doesn't require anything from the client side.

**Comment by pjbroad** [comment]
OK closing.  Again lots of fun. 1.9.7 is coming........
